core/lib/src/fairing/ad_hoc.rs

```diff
@@ -60,35 +60,23 @@ pub struct AdHoc {
 //         Async!($kind ($($param),*) -> ())
 //     );
 // }
-
-/// The attach fairing's function type
-type FnAttach = dyn FnOnce(Rocket)
-    -> BoxFuture<'static, Result<Rocket, Rocket>> + Send + 'static;
-
-/// The launch fairing's function type
-type FnLaunch = dyn FnOnce(&Rocket) + Send + 'static;
-
-/// The request fairing's function type
-type FnRequest = dyn for<'a> Fn(&'a mut Request<'_>, &'a Data)
-    -> BoxFuture<'a, ()> + Send + Sync + 'static;
-
-/// The response fairing's function type
-type FnResponse = dyn for<'a> Fn(&'a Request<'_>, &'a mut Response<'_>)
-    -> BoxFuture<'a, ()> + Send + Sync + 'static;
-
+#[allow(clippy::type_complexity)]
```

Comment:
The [type_complexity](https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity) is configurable inside [clippy.toml](https://github.com/rust-lang/rust-clippy#configuration). You could put a very high number in the configuration file if you want do disable it.